crates/subspace-archiving/src/archiver.rs

```diff
@@ -118,25 +120,44 @@ pub struct ArchivedSegment {
 }
 
 /// Archiver instantiation error
-#[derive(Debug, Error, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+#[cfg_attr(feature = "thiserror", derive(thiserror::Error))]
 pub enum ArchiverInstantiationError {
     /// Record size it smaller that needed to hold any information
-    #[error("Record size it smaller that needed to hold any information")]
+    #[cfg_attr(
+        feature = "thiserror",
```

Comment:
Why not `feature = "std"`?

Reply:
I think it is much nicer to be more specific here. You may potentially have std without thiserror (for thiserror the opposite isn't true, but it is true for serde, so I really like to have a pattern that is applicable in generic way).